### PR TITLE
Randomise promotions intervals

### LINKED ISSUE #12355: Time between promotion checks should be semi-randomized
## Test plan
See https://github.com/brave/browser-laptop/pull/12657

## Original issue description
Currently the promotion check happens exactly every 24 hours. We should introduce a random component.

### PR DESCRIPTION
**Pull request by NejcZdovc**
Resolves #12355

Auditors:

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).
- [ ] Request a security/privacy review [as needed](https://github.com/brave/handbook/blob/master/development/security.md#how-to-request-a-security-review). (Ask a Brave employee to help if you cannot access this document.)

## Test Plan:
- enable ledger
- make sure that promotions is checked on every 20-24 hours

## Reviewer Checklist:

- [ ] Request a security/privacy review [as needed](https://github.com/brave/handbook/blob/master/development/security.md#how-to-request-a-security-review) if one was not already requested.

Tests

- [ ] Adequate test coverage exists to prevent regressions
- [ ] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [ ] New files have MPL2 license header

### CODE REVIEW

File: app/browser/api/ledger.js [hunk 1]
```diff
@@ -1138,6 +1138,19 @@ const initSynopsis = (state) => {
...
+  if (promotionTimeoutId) {
+    clearInterval(promotionTimeoutId)
+  }
+
+  // get promotions
+  appActions.onPromotionGet()
+
+  promotionTimeoutId = setTimeout(() => {
+    checkPromotions()
+  }, random.randomInt({min: 20 * ledgerUtil.milliseconds.second, max: 24 * ledgerUtil.milliseconds.second}))
```
Comment (evq) [inline app/browser/api/ledger.js]: should these be `ledgerUtil.milliseconds.hour`?

Comment (NejcZdovc) [reply app/browser/api/ledger.js]: you are correct, I was using seconds to test code out and forgot to change it back, nice catch

**Comment by bsclifton** [comment]
master 68b5444893e9e4f74629bc311f3930b521e7effa
0.22.x 7257bbfb60866e419effec59a9a8ed3dc29a4ea3
0.21.x e508df05f0d4226cee85ab0eea234f2ea43e9373

**Comment by NejcZdovc** [comment]
0.21 twitch cd1116746

**Pull request merged**